File: src/dataset/cpp/resistor-color-duo_exercism-toemm/gemini/GeminiFlash_resistor_color_duo_v2.cpp
#include "resistor_color_duo.h"

#include <array>

namespace resistor_color_duo {

constexpr std::array<const char*, 10> encoding{"black",  "brown", "red",  "orange",
    "yellow", "green", "blue", "violet",
    "grey",   "white"};

int color_code(const std::string& color) noexcept {
    for (std::size_t i = 0; i < encoding.size(); ++i) {
        if (color == encoding[i]) {
            return i;
        }
    }
    return 0;
}

int value(const std::vector<std::string>& input) noexcept {
    return color_code(input[0]) * 10 + color_code(input[1]);
}

}  // namespace resistor_color_duo

File: src/dataset/Java/resistor-color-duo_exercism-johannesrave-education/gemini/GeminiFlash_ResistorColorDuo_v2.java
import java.util.Map;
import java.util.List;
import java.util.Collections;
import java.util.HashMap;

class ResistorColorDuo {
    private static final Map<String, Integer> colorCodeMap;

    static {
        Map<String, Integer> tempMap = new HashMap<>();
        tempMap.put("black", 0);
        tempMap.put("brown", 1);
        tempMap.put("red", 2);
        tempMap.put("orange", 3);
        tempMap.put("yellow", 4);
        tempMap.put("green", 5);
        tempMap.put("blue", 6);
        tempMap.put("violet", 7);
        tempMap.put("grey", 8);
        tempMap.put("white", 9);
        colorCodeMap = Collections.unmodifiableMap(tempMap);
    }

    int value(String[] colors) {
        return colorCodeMap.get(colors[0]) * 10 + colorCodeMap.get(colors[1]);
    }
}